azure/auth.go

```diff
@@ -0,0 +1,81 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/jenkins-x/go-scm/scm"
+	"github.com/jenkins-x/go-scm/scm/driver/azure"
+	"github.com/jenkins-x/go-scm/scm/transport"
+	"net/http"
+)
+
+const domain = "https://dev.azure.com"
+
+type ClientOptions struct {
+	org     string
+	project string
+	token   string
+}
+
+type wrapper struct {
+	client *scm.Client
+}
+
+func NewClient(clientOptions ClientOptions) (gitprovider.Client, error) {
+
+	project := clientOptions.project
+	org := clientOptions.org
+
+	c, err := azure.New(domain, org, project)
+
+	c.Client = &http.Client{
+		Transport: &transport.Custom{
+			Before: func(r *http.Request) {
+				r.Header.Set("Authorization", fmt.Sprintf("Basic %s", clientOptions.token))
+			},
+		},
+	}
+
+	return wrapper{client: c}, err
+}
+
+// SupportedDomain returns the domain endpoint for this client, e.g. "github.com", "enterprise.github.com" or
+// "my-custom-git-server.com:6443". This allows a higher-level user to know what Client to use for
+// what endpoints.
+// This field is set at client creation time, and can't be changed.
+func (c wrapper) SupportedDomain() string {
+	return domain
+}
+
+// ProviderID returns the provider ID "github".
+// This field is set at client creation time, and can't be changed.
+func (c wrapper) ProviderID() gitprovider.ProviderID {
+	return ""
```

Comment:
You could return https://github.com/jenkins-x/go-scm/blob/main/scm/driver/azure/azure.go#L39 here?